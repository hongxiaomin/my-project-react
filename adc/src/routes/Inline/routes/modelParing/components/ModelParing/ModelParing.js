/* eslint import/no-named-as-default: 0 */
import React from 'react'
import { connect } from 'react-redux'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { Select } from 'antd'
import { assign } from 'lodash'
import classes from './ModelParing.scss'
import ModelSettingList from '../ModelSettingList'
import SearchCondition from '../../../SearchCondition'
import CreateModelParing from '../CreateModelParing'
import { CHANGE_STATUS, GET_MODEL_OPTION,
   getSelectInfo,
   getModelSettingList,
   getModelList,
   MODEL_STATUS,
   CHECK_ERROR } from '../../modules/ModelParing'

type Props = {
  dispatch: Function,
  processList: Array,
  productList: Array,
  modelList: Array,
};


export class ModelParing extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isDialogOpen: false,
      processId: '',
      productId: '',
      modelId: '',
    }
  }

  componentDidMount() {
    this.props.dispatch(getSelectInfo())
    this.props.dispatch(getModelList())
  }
  handleChange = type => (value) => {
    this.setState({
      [type]: value,
    })
  }
  search = () => {
    const tmp = {
      processId: this.state.processId,
      productId: this.state.productId,
      modelId: this.state.modelId,
      queryType: 'list',
    }
    this.props.dispatch(getModelSettingList(tmp))
  }

  toggleDialog = () => {
    this.setState({
      isDialogOpen: !this.state.isDialogOpen,
    })
    this.props.dispatch({
      type: CHANGE_STATUS,
      payload: false,
    })
    this.props.dispatch({
      type: GET_MODEL_OPTION,
      payload: [],
    })
    this.props.dispatch({
      type: MODEL_STATUS,
      payload: '',
    })
    this.props.dispatch({
      type: CHECK_ERROR,
      payload: '',
    })
  }
  props: Props
  render() {
    const { processId, productId, modelId } = this.state
    const filter = []
    this.props.modelList.forEach((item) => {
      if (item.isPublished === true) {
        filter.push(item)
      }
    })
    const model = filter.map(item => ((item.isDefault) ? assign({}, item, { name: `${item.name}(Default Model)` }) : item))
    return (
      <div>
        <div className={classes.container}>
          <div className={classes.top}>
            <h4 className={classes.model}><WrapperFormattedMessage id='modelPairing.modelParing.modelPairingSetting' /></h4>
            <button
              style={{ width: '80px' }}
              type='button'
              onClick={this.toggleDialog}
              className={'pt-button pt-intent-primary'}
            >
              <WrapperFormattedMessage id='create' />
            </button>
          </div>
          <SearchCondition
            processList={this.props.processList}
            productList={this.props.productList}
            modelList={model}
            handleChange={this.handleChange}
            searchChange={this.search}
            isClick={false}
          />
        </div>
        <ModelSettingList
          modelList={this.props.modelList}
          searchParam={{
            processId,
            productId,
            modelId,
          }}
        />
        <CreateModelParing
          toggleDialog={this.toggleDialog}
          isOpen={this.state.isDialogOpen}
        />
      </div>
    )
  }
}
export default connect(
  ({ auth, modelParing, intl }) => ({
    intl,
    token: auth.token,
    processList: modelParing.processList,
    productList: modelParing.productList,
    modelOption: modelParing.modelOption,
    modelList: modelParing.modelList,
  }),
)(ModelParing)
