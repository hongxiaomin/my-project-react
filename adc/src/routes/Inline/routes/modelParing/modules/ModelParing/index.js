import { assign, filter, isEmpty } from 'lodash'
import { showMessage, stringToFirstUpperCase } from 'Utils'
import { simpleFetch } from 'helpers/fetchWrapper'


// ------------------------------------
// Constants
// ------------------------------------
export const PRODUCT_PROCESS_DEFECTCODE = 'PRODUCT_PROCESS_DEFECTCODE'
export const CHECK_ERROR = 'CHECK_ERROR'
export const GET_MODEL_SETTING_LIST = 'GET_MODEL_SETTING_LIST'
export const GET_MODEL_OPTION = 'GET_MODEL_OPTION'
export const MODEL_LIST = 'MODEL_LIST'
export const GET_MODEL_DEFECT = 'GET_MODEL_DEFECT'
export const SELECT_DATA = 'SELECT_DATA'
export const GET_EDIT_MODEL_OPTION = 'GET_EDIT_MODEL_OPTION'
export const CHANGE_STATUS = 'CHANGE_STATUS'
export const DELETE_MODEL = 'DELETE_MODEL'
export const MODEL_STATUS = 'MODEL_STATUS'
// ------------------------------------
// Async actions
// ------------------------------------
export const getSelectInfo = () => (dispatch) => {
  // 请求获取product, process, defectcode 信息
  simpleFetch('labeling/image/search')
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ text: stringToFirstUpperCase(data.message), type: 'error' })
        return
      }
      const result = data.result
      dispatch({
        type: PRODUCT_PROCESS_DEFECTCODE,
        data: result,
      })
    })
}
export const getModelList = () => (dispatch) => {
    // 请求获取model
  simpleFetch('models')
  .then((data) => {
    if (data.code === 200000) {
      dispatch({
        type: MODEL_LIST,
        payload: data.result,
      })
    }
  })
}
export const getModelDefect = (id) => (dispatch) => {
    // simpleFetch('/api/defectCode')
  simpleFetch(`model/defect?modelId=${id}`)
  // .then(response => response.json())
  .then((data) => {
    if (data.code === 200000) {
      dispatch({
        type: GET_MODEL_DEFECT,
        data: data.result,

      })
    }
  })
  .catch(err => console.log(err))
}
export const getModelSettingList = data => (dispatch) => {
  // search接口，任选一个条件
  let processId = data.processId
  let productId = data.productId
  let modelId = data.modelId
  if (isEmpty(data.processId)) {
    processId = ''
  }
  if (isEmpty(data.productId)) {
    productId = ''
  }
  if (isEmpty(data.modelId)) {
    modelId = ''
  }
  simpleFetch(`judging/configuration?queryType=${data.queryType}&` +
    `processId=${processId}&productId=${productId}&modelId=${modelId}`)
  .then((json) => {
    if (json.code === 200000) {
      dispatch({
        type: GET_MODEL_SETTING_LIST,
        payload: JSON.parse(json.result),
      })
    } else {
      // dispatch({
      //   type: GET_MODEL_SETTING_LIST,
      //   payload: json.result.data,
      // })
    }
  })
}
export const addProcessProduct = state => (dispatch) => {
  // search接口，任选一个条件
  simpleFetch('judging/configuration', {
    method: 'POST',
    body: state,
  })
  .then((json) => {
    if (json.code === 200000) {
      dispatch(getModelSettingList({ queryType: 'list' }))
      showMessage({ text: json.message, type: 'success' })
    } else {
      showMessage({ text: json.message, type: 'error' })
    }
  })
}
export const updateProcessProduct = (state, cb) => () => {
  simpleFetch('judging/configuration', {
    method: 'PUT',
    body: state,
  })
  .then((json) => {
    if (json.code === 200000) {
      showMessage({ text: json.message, type: 'success' })
      if (cb) cb()
    } else {
      showMessage({ text: json.message, type: 'error' })
    }
  })
}
// export const getSelectData = () => (dispatch, getState) =>
//     // simpleFetch(`${__URL__}/ojs/services`, {
//     simpleFetch('/api/ojs/selectData', {
//       headers: {
//         'Content-Type': 'application/json',
//         Accept: 'application/json',
//         Authorization: getState().auth.token,
//       },
//     })
//     .then((json) => {
//       if (json.code === 200000) {
//         dispatch({
//           type: SELECT_DATA,
//           payload: json.result.data,
//         })
//       } else {
//         dispatch({
//           type: SELECT_DATA,
//           payload: json.result.data,
//         })
//       }
//     })
export const getModelOption = (processId, productId) => (dispatch) => {
  // 根据process和product校验model
  simpleFetch(`judging/configuration/check?processId=${processId}&productId=${productId}`)
  .then((json) => {
    if (json.code === 200000) {
      dispatch({
        type: GET_MODEL_OPTION,
        payload: JSON.parse(json.result),
      })
      dispatch({
        type: CHANGE_STATUS,
        payload: true,
      })
    } else {
      dispatch({
        type: CHECK_ERROR,
        payload: json.message,
      })
    }
  })
}
export const deleteModel = id => dispatch =>
        // simpleFetch(`${__URL__}/ojs/services`, {
  simpleFetch('judging/configuration', {
    method: 'DELETE',
    body: { configId: id.toString() },
  })
  .then((json) => {
    if (json.code === 200000) {
      dispatch({
        type: DELETE_MODEL,
        payload: { configId: id },
      })
      showMessage({ text: json.message, type: 'success' })
    } else {
      showMessage({ text: json.message, type: 'error' })
    }
  })

const ACTION_HANDLERS = {
  [PRODUCT_PROCESS_DEFECTCODE]: (state, action) => {
    return Object.assign(
      {},
      state,
      {
        // defectCodeList: action.data.defect,
        productList: action.data.product,
        processList: action.data.process,
      },
    )
  },
  [GET_MODEL_SETTING_LIST]: (state, action) => assign({}, state, {
    modelSettingList: action.payload,
  }),
  [MODEL_LIST]: (state, action) => assign({}, state, {
    modelList: action.payload,
  }),
  [CHECK_ERROR]: (state, action) => assign({}, state, {
    checkError: action.payload,
  }),
  [GET_MODEL_OPTION]: (state, action) => {
    return assign({}, state, {
      modelOption: action.payload,
    })
  },
  [SELECT_DATA]: (state, action) => assign({}, state, {
    selectData: action.payload,
  }),
  [GET_EDIT_MODEL_OPTION]: (state, action) => assign({}, state, {
    editModelOption: action.payload,
  }),
  [DELETE_MODEL]: (state, action) => {
    const { configId } = action.payload
    return assign({}, state, {
      modelSettingList: filter(state.modelSettingList, el => el.configId * 1 !== configId * 1),
    })
  },
  [CHANGE_STATUS]: (state, action) => assign({}, state, {
    status: action.payload,
  }),
  [MODEL_STATUS]: (state, action) => assign({}, state, {
    modelstatus: action.payload,
  }),
  [GET_MODEL_DEFECT]: (state, action) => {
    return assign({}, state, {
      modelDefect: action.data
    })
  },
}


const initialState = {
  modelSettingList: [],
  modelOption: [],
  checkError: '',
  editModelOption: [],
  status: false,
  selectData: [],
  modelstatus: '',
  productList: [],
  processList: [],
  modelList: [],
}

export const modelParingReducer = (state = initialState, action) => {
  const handler = ACTION_HANDLERS[action.type]

  return handler ? handler(state, action) : state
}
