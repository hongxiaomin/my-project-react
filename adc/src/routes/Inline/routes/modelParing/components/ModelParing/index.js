import ModelParing from './ModelParing'

export default ModelParing
