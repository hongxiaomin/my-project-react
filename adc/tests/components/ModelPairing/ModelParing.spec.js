import React from 'react'
import { ModelParing } from 'routes/Inline/routes/ModelParing/components/ModelParing/ModelParing'

describe('(Component) ModelParing', () => {
  const processList = []
  const productList = []
  const modelList = [{ id: 1, isPublished: true, isDefault: true }, { id: 1, isPublished: true, isDefault: false }]
  const dispatch = jest.fn()
  it('render normally', () => {
    const wrapper = shallow(
      <ModelParing
        processList={processList}
        productList={productList}
        modelList={modelList}
        dispatch={dispatch}
      />,
    )
    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.state('isDialogOpen')).toEqual(true)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('Select').at(0).simulate('change', 'aa')
    expect(wrapper.state('processId')).toEqual('aa')
    wrapper.find('Select').at(1).simulate('change', 'bb')
    expect(wrapper.state('productId')).toEqual('bb')
    wrapper.find('Select').at(2).simulate('change', 'cc')
    expect(wrapper.state('model')).toEqual('cc')
    expect(wrapper).toMatchSnapshot()
  })
})
