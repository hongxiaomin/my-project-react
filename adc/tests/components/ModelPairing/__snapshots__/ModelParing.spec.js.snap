// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ModelParing render normally 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="top"
    >
      <h4
        className="model"
      >
        Model Pairing Setting
      </h4>
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
        style={
          Object {
            "width": "80px",
          }
        }
        type="button"
      >
        Create
      </button>
    </div>
    <div
      className="searchDiv"
    >
      <h5
        style={
          Object {
            "marginRight": "10px",
            "marginTop": "6px",
          }
        }
      >
        Search condition:
      </h5>
      <div
        className="singleSearch"
      >
        <h5
          className="h5"
        >
          Process ID
        </h5>
        <Select
          allowClear={true}
          choiceTransitionName="zoom"
          name="processId"
          onChange={[Function]}
          optionFilterProp="children"
          placeholder="Please select"
          prefixCls="ant-select"
          showSearch={true}
          style={
            Object {
              "width": 200,
            }
          }
          transitionName="slide-up"
        />
      </div>
      <div
        className="singleSearch"
      >
        <h5
          className="h5"
        >
          Product ID
        </h5>
        <Select
          allowClear={true}
          choiceTransitionName="zoom"
          name="processId"
          onChange={[Function]}
          optionFilterProp="children"
          placeholder="Please select"
          prefixCls="ant-select"
          showSearch={true}
          style={
            Object {
              "width": 200,
            }
          }
          transitionName="slide-up"
        />
      </div>
      <div
        className="singleSearch"
      >
        <h5
          className="h5"
        >
          Model
        </h5>
        <Select
          allowClear={true}
          choiceTransitionName="zoom"
          name="processId"
          onChange={[Function]}
          optionFilterProp="children"
          placeholder="Please select"
          prefixCls="ant-select"
          showSearch={true}
          style={
            Object {
              "width": 200,
            }
          }
          transitionName="slide-up"
        >
          <Option
            value="1"
          >
            undefined(Default Model)
          </Option>
          <Option
            value="1"
          />
        </Select>
      </div>
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
        style={
          Object {
            "width": "80px",
          }
        }
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  <Connect(ModelSettingList)
    modelList={
      Array [
        Object {
          "id": 1,
          "isDefault": true,
          "isPublished": true,
        },
        Object {
          "id": 1,
          "isDefault": false,
          "isPublished": true,
        },
      ]
    }
    searchParam={
      Object {
        "modelId": "cc",
        "processId": "aa",
        "productId": "bb",
      }
    }
  />
  <Connect(CreateModelParing)
    isOpen={true}
    toggleDialog={[Function]}
  />
</div>
`;
